hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/handler/RegionReplicaFlushHandler.java

```diff
@@ -130,11 +130,7 @@ void triggerFlushInPrimaryRegion(final HRegion region) throws IOException {
           ServerRegionReplicaUtil.getRegionInfoForDefaultReplica(region.getRegionInfo())
             .getRegionNameAsString(),
           region.getRegionInfo().getRegionNameAsString(), counter.getAttemptTimes(), e);
-        try {
```

Comment:
You did not get what I was saying I believe.
This is triggerFlushInPrimaryRegion() method and u can see a while loop within which the call happening.  The while loop is to be terminated once this RS is set to be stopped/abort.  When such happens, there might be many non daemon threads running within the server.  Our logic in different places will interrupt these threads.  so if the thread is sleeping or waiting it will get InterruptedException and allow the thread NOT to continue in running/waiting state. The logic should be checking the server status and allow to come out of loops etc. 
But your change will make it such that even if the main thread interrupt this thread, it will continue to sleep for the specified time.  That is totally against our intent.

Reply:
Ok got it. Yes this example makes sense. Are you saying that similar to this one, all the other places also need to handle Interruptions?

Reply:
I did not check all places.. Most of the places where the RetryCounter is being used is in test I can see.  But within RetryCounter  we should not change. Tomorrow some other code path might use it too.  IMO we can just keep the code as is. Let the calling part handle the InterruptedException the way they want. we can not generalise it.    So just close this Jira Viraj

Reply:
Okk just saw that many places are in asynchronously getting executed in separate threads from main thread and if this is how interruptions were planned to be handled, it's fine. No need to make change.